src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,197 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.task.*;
+import duke.storage.Storage;
+import duke.ui.Ui;
+
+/**
+ * Executes a command based on parsed inputs.
+ */
+public class Command {
+    private String commandName;
+    private String taskName;
+    private String by;
+    private String from;
+    private String to;
+    private int index;
+    private boolean isExit = false;
+
+
+    private enum CommandName {
+        BYE, LIST, MARK, UNMARK, DELETE, TODO, DEADLINE, EVENT, FIND
+    }
+
+    /**
+     * Constructor for commands 'bye' and 'list'.
+     *
+     * @param commandName The name of the command.
+     */
+    public Command(String commandName) {
+        this.commandName = commandName;
+    }
+
+    /**
+     * Constructor for commands "mark", "unmark" and "delete".
+     *
+     * @param commandName The name of the command.
+     * @param index The index of the task in taskList to be executed on.
+     */
+    public Command(String commandName, int index) {
+        this.commandName = commandName;
+        this.index = index;
+    }
+
+    /**
+     * Constructor for command "todo" and "find".
+     *
+     * @param commandName The name of the command.
+     * @param taskName The name of the todo task.
+     */
+    public Command(String commandName, String taskName) {
+        this.commandName = commandName;
+        this.taskName = taskName;
+    }
+
+    /**
+     * Constructor for command "deadline".
+     *
+     * @param commandName The name of the command.
+     * @param taskName The name of the deadline task.
+     * @param by The date of the deadline, must be in yyyy-mm-dd format.
+     */
+    public Command(String commandName, String taskName, String by) {
+        this.commandName = commandName;
+        this.taskName = taskName;
+        this.by = by;
+    }
+
+    /**
+     * Constructor for command "deadline".
+     *
+     * @param commandName The name of the command.
+     * @param taskName The name of the event task.
+     * @param from The event start date.
+     * @param to The event end date.
+     */
+    public Command(String commandName, String taskName, String from, String to) {
+        this.commandName = commandName;
+        this.taskName = taskName;
+        this.from = from;
+        this.to = to;
+    }
+
+    /**
+     * Executes the given command.
+     * Throws exception if the argument is not recognized or in a wrong format.
+     *
+     * @param tasks TaskList object containing the list of tasks
+     * @param ui The Ui object to display messages.
+     * @param storage The Storage object to save the task after execution.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        CommandName cn = CommandName.valueOf(this.commandName.toUpperCase());
```

Comment:
Perhaps you could consider removing the this keyword here and the rest of the code to remove unecessary noise

```suggestion
        CommandName cn = CommandName.valueOf(commandName.toUpperCase());
```